### ISSUE TITLE
Participant: institution should remain a text field after autocompletion

### ISSUE THREAD

**Issue by cecilia-donnelly** [OpenTechStrategies/swoptact#78]
When adding an institution to a Participant, don't display the Institution name with a gray "x" as the autocomplete_light library does by default.  Instead, continue showing it as a text field.  But we don't allow arbitrary text here -- it must match an existing institution.

**Comment by cecilia-donnelly** [comment]
This will probably be fixed by @kfogel's change to the autocomplete form in #94.  That is, I expect whatever change he makes will take effect across all autocomplete fields.

**Comment by cecilia-donnelly** [comment]
@nttaylor since you did some of this detective work in #123, maybe you have some ideas of how to do this?  If not, then talk to @cwebber and @xray7224.

**Comment by cwebber** [comment]
So, I just committed some progress; it isn't nearly there yet.  It has taken me quite a long time to figure out the right direction for this.  Once again I looked heavily into using django_autocomplete_light's extensions, but it turned out to be overwhelming... there was so much metaclassing and magic that I couldn't keep on top of it.

The good news is that solving this should not be so hard from this point forward; it took me some time to become reaccquainted with how django modelforms work (and how they have changed), but I have most components in the right direction.

Remaining tasks:
 - Move this to the actually correct model and field
 - Get the autocomplete widget to have the right initial value
 - Save data correctly on page save
 - Add UI to widget, and hooking up to backend to query whether or not an entry is new or not and displaying that (see below for thoughts on how this will look)

As for my thoughts on how this functions...

My thought is that this will simply be filling in a text version of whatever this is linked to with a foreign key, and on save, changing that foreign key to whatever matches text entered into the box (I have no suggestions on how to resolve a naming conflict, however).  The interface for checking whether something matches or not will be very similar to in the "inline ajax'y interface", however since *UNLIKE* the inline row ajax'y many to many editing interface completing on instiutions, we will *not* be adding new institutions when the user types something new, we need to give clear indication on whether input is acceptable or not (ie, matches an existing institution).

To that end, I suggest an interface like the following:

```
[programmer______] [ OK! ✓ ]     <-- green, meaning we have a match
[nogrammar_______] [ ✗ ]         <-- red, meaning not acceptable / no match
```

**Comment by cecilia-donnelly** [comment]
As @cwebber and I discussed this morning, we are allowing users to create new institutions from this interface, so we don't need to worry about blocking incorrect inputs.

So, the interface we're going for here is: input box is visible, users type into it, get autocomplete options and either choose one or continue typing a new object name, and then save.  After save, the institution name is still visible in an input box.

**Comment by cwebber** [comment]
Good news everyone!  The widget works, and you can complete it and save it and stuff.

However, we talked about being able to create new things when the user provides a value that is not provided... I forgot about this, so I need to add that, if it's still wanted.

Nonetheless, new functionality added to master!  Should be easy to drop into other places, too...

**Comment by cwebber** [comment]
Leaving this open for convo w/ cdonnelly tomorrow morning!

**Comment by cecilia-donnelly** [comment]
Hey, @cwebber -- thanks for the notes!  Unfortunately, I'm getting an error when I try to run this locally:

```
Traceback (most recent call last):
  File "manage.py", line 26, in <module>
    execute_from_command_line(sys.argv)
  File "/swoptact/lib/python3.4/site-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
    utility.execute()
  File "/swoptact/lib/python3.4/site-packages/django/core/management/__init__.py", line 312, in execute
    django.setup()
  File "/swoptact/lib/python3.4/site-packages/django/__init__.py", line 18, in setup
    apps.populate(settings.INSTALLED_APPS)
  File /swoptact/lib/python3.4/site-packages/django/apps/registry.py", line 115, in populate
    app_config.ready()
  File /swoptact/lib/python3.4/site-packages/django/contrib/admin/apps.py", line 22, in ready
    self.module.autodiscover()
  File "/swoptact/lib/python3.4/site-packages/django/contrib/admin/__init__.py", line 24, in autodiscover
    autodiscover_modules('admin', register_to=site)
  File "/swoptact/lib/python3.4/site-packages/django/utils/module_loading.py", line 74, in autodiscover_modules
    import_module('%s.%s' % (app_config.name, module_to_search))
  File "/swoptact/lib/python3.4/importlib/__init__.py", line 109, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1471, in exec_module
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "/swoptact/swoptact/admin.py", line 416, in <module>
    class EventAdmin(mixins.AdminArchiveMixin, AjaxyInlineAdmin):
  File "/swoptact/swoptact/admin.py", line 424, in EventAdmin
    form=st_forms.TagSkippingAutoCompleteModelForm,
AttributeError: 'module' object has no attribute 'TagSkippingAutoCompleteModelForm'
```

I did a `dev-refresh` so presumably I should have all the correct settings, if they needed to be changed.  I'll do some more poking around and report back here when I figure out what's going on.

**Comment by cecilia-donnelly** [comment]
Ah, it looks like the last commit, e2d743e29dcd03163ef1c24ea81f6f760315a412, must have removed something that should have stayed in.  When I check out the previous commit, 80aa055f426793127007a3e6584c0137a8e6e91d, everything works fine.

**Comment by cecilia-donnelly** [comment]
After further investigation:

In e2d743e, we removed the creation of the `TagSkippingAutocompleteModelForm` from `forms.py`, but it is still referred to several times in `admin.py`.  I removed it from `admin.py` in 9d290bf493e2f8e97c6e8c2b2cdc3ce89486789f and everything seems to work fine.  @cwebber, does that sound right?  I assume that we don't need it anymore in `admin.py`, but let me know if I should replace it with something instead of just removing it outright.
